### DISCUSSION TITLE
Using MET to calculate 500mb ACC scores

### DISCUSSION THREAD

**Discussion by awhuffma** [dtcenter/METplus#1113] [Statistical Computation] [answered]
Good Day,

I have for years used MET to calculate 500 mb Height ACC scores from gridded model output. However, recently I upgraded from version 8.1.2 to 9.1.3 and ran into an issue. 

I have previously used grib1 versions of:
Forecast File GFS (0.5 degree)
gfs.t00z.pgrb2.0p50.f000
Observation File (GDAS analysis 0.25 degree:
gdas.t00z.pgrb2.0p25.f000
Climo File (CFSR converted to grib1)
https://cfs.ncep.noaa.gov/pub/raid0/cfsv2/climo_cfsr_time/mean/z500.cfsr.mean.clim.daily.1982.2010.grb2

I have attached by GridStatConfig file and also pasted below my output. Any help would be appreciated. Thanks.

This is the output I get using 9.1.3:
DEBUG 1: Default Config File: /usr/local/share/met/config/GridStatConfig_default
DEBUG 1: User Config File: GridStatConfig_default
ERROR  : 
ERROR  : check_met_version() -> The version number listed in the config file (V8.1.2) is not compatible with the current version of the code (V9.1.3).
ERROR  : 
[ahuffman@dev01]$ vi GridStatConfig_default
[ahuffman@dev01]$ grid_stat gfs.t00z.pgrb.0p50.f000 /cfs/devA/home/ahuffman/gdas/2021082500gdas.grib GridStatConfig_default -outdir .
DEBUG 1: Default Config File: /usr/local/share/met/config/GridStatConfig_default
DEBUG 1: User Config File: GridStatConfig_default
WARNING: 
WARNING: GridStatVxOpt::process_config() -> "nc_pairs_var_str" ((nul)) is deprecated and replaced by "nc_pairs_var_suffix".
WARNING: 
GSL_RNG_TYPE=mt19937
GSL_RNG_SEED=18446744073233639009
DEBUG 1: Forecast File: gfs.t00z.pgrb.0p50.f000
DEBUG 1: Observation File: /cfs/devA/home/ahuffman/gdas/2021082500gdas.grib
DEBUG 1: Regridding field HGT/P500 to the verification grid.
DEBUG 1: Regridding the 19840726_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840727_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840728_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840729_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840730_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840731_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840801_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840802_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840803_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840804_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840805_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840806_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840807_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840808_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840809_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840810_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840811_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840812_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840813_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840814_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840815_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840816_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840817_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840818_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840819_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840820_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840821_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840822_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840823_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840824_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840825_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840826_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840827_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840828_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840829_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840830_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840831_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840901_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840902_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840903_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840904_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840905_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840906_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840907_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840908_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840909_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840910_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840911_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840912_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840913_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840914_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840915_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840916_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840917_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840918_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840919_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840920_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840921_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840922_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840923_000000 "HGT/P500" climatology field to the verification grid.
DEBUG 1: Regridding the 19840924_000000 "HGT/P500" climatology field to the verification grid.
ERROR  : 
ERROR  : climo_hms_interp() -> Can't find climatology data before/after date 19700101_051424.
ERROR  : 
[GridStatConfig_default.docx](https://github.com/dtcenter/METplus/files/7049679/GridStatConfig_default.docx)

**Comment by j-opatz** [comment]
Hello, and thank you for your question. 

I took a look at the grib2 file you attached and confirmed that the dates of the grib2 file match your output. From the output and error provided, it would seem like one of the entries in the grib2 file, most likely 19840925 for the HGT/P500 entry, has a valid date being read by MET as a bad date, so it's defaulting to 19700101. 

I'm going to include @JohnHalleyGotway on this issue as he should be able to provide better insight on what's causing the `climo_hms_interp` error.

**Comment by JohnHalleyGotway** [comment, marked as answer]
@awhuffma, thanks for sending the sample climatology data file. I took a look at the records in there and do NOT see any issues there. To summarize, the z500.cfsr.mean.clim.daily.1982.2010.grb2 file contains 1464 GRIB2 records, all for 500mb height. The records are spaced 6 hours apart during 1984 (4 per day for 366 days including 2/29).

You're running Grid-Stat to compare gfs.t00z.pgrb.0p50.f000 to 2021082500gdas.grib. However, as the log message indicates, you're using an old configuration file version:
```
ERROR : check_met_version() -> The version number listed in the config file (V8.1.2) is not compatible with the current version of the code (V9.1.3).
```
When the major version number of the code (i.e. 9) differs from the major number listed in the config file (i.e. 8), we print that error message to alert users that the configuration options may have changed. And between MET versions 8.1.2 and 9.1.3, the way you configure information about climatology data changed. I see that you simply changed the version number listed in the config file without actually changing the contents.

In general, I'd recommend diffing your v8.1.2 configuration file against the default Grid-Stat config file for v9.1.3. It should be pretty easy to see what's changed and then you can migrate the specifics of your v8.1.2 configuration into a file for v9.1.3.

Comparing the default Grid-Stat config file between versions [8.1.2](https://github.com/dtcenter/MET/blob/v8.1.2/met/data/config/GridStatConfig_default) and [9.1.3](https://github.com/dtcenter/MET/blob/v9.1.3/met/data/config/GridStatConfig_default), you'll find that inside the "climo_mean" dictionary:
```
   -- match_day          = FALSE; // WAS REMOVED
   -- time_step          = 21600; // WAS REMOVED
```
```
   ++ day_interval       = 31; // WAS ADDED
   ++ hour_interval      = 6; // WAS ADDED
```
You can read more about the climo_mean settings in [this section](https://met.readthedocs.io/en/develop/Users_Guide/config_options.html#climo-mean) of the MET User's Guide.

But for this data, that is daily and spaced 6-hours apart, you should use:
```
day_interval = 1;
hour_interval = 6;
```
Do note that there are other changes between the Grid-Stat config file between these versions. So I'd recommend looking more closely at those.

I'm not sure this will fix the error you're getting, but we should definitely update the config file before debugging any further. Please let me know if/how the behavior changes after you update the config file.

I do realize this is a hassle and apologize for that. Whenever possible, we do make config file changes backward compatible. But in this case, doing so confused the logic of the code considerably.

**Reply by awhuffma** [reply]
John,

Thanks so much for the reply. This did fix my issue after I updated my config file. Should have thought of that myself. Thanks again for the help!

Allan
